test/unit/uploader.basic.api.js

```diff
@@ -479,4 +479,62 @@ describe("uploader.basic.api.js", function () {
             uploader._handleNewFile(fileInput, 0, []);
         });
     });
+
+    describe("_formatSize", function() {
+        beforeEach(function () {
+            fineuploader = new qq.FineUploaderBasic();
+        });
+
+        it("formats 0 bytes properly", function() {
+            var formattedSize = fineuploader._formatSize(0);
+            assert.equal(formattedSize, "0kB");
+        });
+
+        it("formats kB properly", function() {
+            var formattedSize = fineuploader._formatSize(789);
+            assert.equal(formattedSize, "0.8kB");
+        });
+
+        it("formats MB properly", function() {
+            var formattedSize = fineuploader._formatSize(2123456);
+            assert.equal(formattedSize, "2.1MB");
+        });
+
+        it("formats GB properly", function() {
+            var formattedSize = fineuploader._formatSize(9602123456);
+            assert.equal(formattedSize, "9.6GB");
+        });
+    });
+
+    describe("_validateFileOrBlobData", function() {
+        var originalFileOrInput = qq.isFileOrInput;
+        beforeEach(function () {
+            fineuploader = new qq.FineUploaderBasic();
+        });
+        afterEach(function() {
+            qq.isFileOrInput = originalFileOrInput;
+        });
+
+        it("fails if file is empty and allowEmpty is false", function() {
+            var result = 1;
+            qq.isFileOrInput = function() { return true; };
+            fineuploader._fileOrBlobRejected = function() { result = -1; };
+            var validationDescriptor = { size: 0 };
+
+            fineuploader._validateFileOrBlobData({}, validationDescriptor)
+                .done(function() { assert.equal(-1, result); });
```

Comment:
Please ignore this comment. I understand what you mean now that I've looked up your example and your mocha reference properly